### ISSUE TITLE
Should output error for `Callable[[T], [T]]`

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1386]
I wrote a wrong code like the below:

```python
def foo(f: Callable[[T], [T]]): # <- return type [T] is incorrect
    ...
```

and this code raises runtime error:

```
TypeError: Callable[args, result]: result must be a type. Got [~T].
```

But pyright doesn't output any error.
I think pyright should output error for this mistake.

**Comment by erictraut** [comment]
Thanks for the report. This will be addressed in the next release of pyright.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.103, which I just published. It will also be included in the next release of pylance.
